common/djangoapps/third_party_auth/utils.py

```diff
@@ -0,0 +1,13 @@
+from random import randint
+
+from django.contrib.auth.models import User
+
+
+def generate_username(username):
+	validate_user = User.objects.filter(username=username).exists()
+	while validate_user:
```

Comment:
Suppose `validate_user=True`, this code would generate an infinite loop. Also, whenever we suggest a `new_username` we must validate once again is not already on DB.